packages/overlay/src/OverlayBase.ts

```diff
@@ -0,0 +1,893 @@
+            });
+        });
+    });
+    const handleTransitionend = (event: TransitionEvent): void => {
+        if (event.propertyName === 'visibility') {
+            // Ignore "visibility" transitions because they often happen before/after a
+            // larger transition and don't represent the overall transition duration.
+            return;
+        }
+        cleanup();
+    };
+    const handleTransitionrun = (event: TransitionEvent): void => {
+        if (event.propertyName === 'visibility') {
+            // Ignore "visibility" transitions because they often happen before/after a
+            // larger transition and don't represent the overall transition duration.
+            return;
+        }
+        cancelAnimationFrame(guarantee1);
+        cancelAnimationFrame(guarantee2);
+        cancelAnimationFrame(guarantee3);
+        el.removeEventListener('transitionrun', handleTransitionrun);
+        el.addEventListener('transitionend', handleTransitionend);
+    };
+    el.addEventListener('transitionrun', handleTransitionrun);
+    action();
+};
+
+export class OverlayBase extends SpectrumElement {
+    static override styles = [styles];
+
+    @property({ type: Boolean })
+    delayed = false;
+
+    @query('.dialog')
+    dialogEl!: HTMLDialogElement & {
+        showPopover(): void;
+        hidePopover(): void;
+    };
+
+    @property({ type: Boolean })
+    get disabled(): boolean {
+        return this._disabled;
+    }
+
+    set disabled(disabled: boolean) {
+        this._disabled = disabled;
+        if (disabled) {
+            if (this.hasNonVirtualTrigger) {
+                this.unbindEvents(this.triggerElement as HTMLElement);
+            }
+            this.wasOpen = this.open;
+            this.open = false;
+        } else {
+            this.bindEvents(this.triggerElement);
+            this.open = this.open || this.wasOpen;
+            this.wasOpen = false;
+        }
+    }
+
+    private _disabled = false;
+
+    protected dispose = noop;
+
+    @queryAssignedElements({
+        flatten: true,
+        selector: ':not([slot="longpress-describedby-descriptor"])', // gather only elements slotted into the default slot
+    })
+    elements!: OpenableElement[];
+
+    public parentOverlayToForceClose?: OverlayBase;
+
+    private get hasNonVirtualTrigger(): boolean {
+        return (
+            !!this.triggerElement &&
+            !(this.triggerElement instanceof VirtualTrigger)
+        );
+    }
+
+    protected longpressed = false;
+
+    private longressTimeout!: ReturnType<typeof setTimeout>;
+
+    @property({ type: Number })
+    offset: number | [number, number] = 6;
+
+    public placementController = new PlacementController(this);
+
+    @property({ type: Boolean, reflect: true })
+    get open(): boolean {
+        return this._open;
+    }
+
+    set open(open: boolean) {
+        if (open && this.disabled) return;
+        if (open === this.open) return;
+        this._open = open;
```

Comment:
Would this be a better place to put `OverlayBase.openCount += 1` than `manageOpen`, in that it's a more reliable place to track the ultimate opening & closing of overlays?

(unless I misunderstand what `openCount` is for)